pkg/variantregistry/ocp_test.go

```diff
@@ -0,0 +1,333 @@
+package variantregistry
+
+import (
+	"testing"
+
+	"github.com/openshift/sippy/pkg/testidentification"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVariantSyncer(t *testing.T) {
+	variantSyncer := OCPVariantLoader{VariantManager: testidentification.NewOpenshiftVariantManager()}
+	tests := []struct {
+		job          string
+		variantsFile map[string]string
+		expected     map[string]string
+	}{
+		{
+			job: "periodic-ci-openshift-release-master-nightly-4.16-e2e-gcp-ovn-fips",
+			variantsFile: map[string]string{
+				"Foo":         "bar",          // should be added
+				"CloudRegion": "us-central-1", // should be ignored
+			},
+			expected: map[string]string{
+				VariantRelease:          "4.16",
+				VariantFromRelease:      "4.16",
+				VariantReleaseMajor:     "4",
+				VariantReleaseMinor:     "16",
+				VariantFromReleaseMajor: "4",
+				VariantFromReleaseMinor: "16",
+				VariantArch:             "amd64",
+				VariantInstaller:        "ipi",
+				VariantFeatureSet:       "default",
+				VariantPlatform:         "gcp",
+				VariantNetwork:          "ovn",
+				VariantNetworkAccess:    "default",
+				VariantOwner:            "eng",
+				VariantScheduler:        "default",
+				VariantTopology:         "ha",
+				VariantSecurityMode:     "fips",
+				VariantSuite:            "unknown",
+				VariantUpgrade:          "none",
+				"Foo":                   "bar",
+			},
+		},
+		{
+			job: "periodic-ci-openshift-hypershift-release-4.16-periodics-e2e-aws-ovn-conformance",
+			expected: map[string]string{
+				VariantRelease:          "4.16",
+				VariantFromRelease:      "4.16",
+				VariantReleaseMajor:     "4",
+				VariantReleaseMinor:     "16",
+				VariantFromReleaseMajor: "4",
+				VariantFromReleaseMinor: "16",
+				VariantArch:             "amd64",
+				VariantInstaller:        "ipi", // Is this ok for hypershift?
```

Comment:
Is it IPI? I don't think it uses the openshift-installer does it?

I'm not sure how to show hypershift, but we probably do need to differentiate it somehow.  For example, we'll probably want to show hypershift on aws and hypershift on metal. Not sure if that's a case for hyphenation (hypershift-aws) or a new category

Reply:
I'll find out, iirc they use a custom installer binary.

Reply:
With regards to platform I think it would be ControlPlane=external, Platform=aws

Reply:
Does that always mean hypershift? I'm just trying to imagine how we would show a view in component readiness where we could look at just hypershift for example

Reply:
ControlPlane external means hypershift i think yes, but per the installer comment we'll also have Installer=Hypershift.